geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py

```diff
@@ -0,0 +1,152 @@
+                models.update(
+                    {
+                        "reference_model": self.pseudo3d_params.reference_model,
+                        "lower_bound": self.pseudo3d_params.lower_bound,
+                        "upper_bound": self.pseudo3d_params.upper_bound,
+                    }
+                )
+
+            for uuid, trial in lookup.items():
+
+                status = trial.pop("status")
+                if status == "pending":
+                    filepath = os.path.join(self.working_directory, f"{uuid}.ui.geoh5")
+                    with Workspace(filepath) as iter_workspace:
+
+                        receiver_entity = extract_dcip_survey(
+                            iter_workspace,
+                            data.entity,
+                            self.pseudo3d_params.line_object.values,
+                            trial["line_id"],
+                        )
+
+                        mesh = get_drape_model(
+                            iter_workspace,
+                            "Models",
+                            receiver_entity.vertices,
+                            [
+                                self.pseudo3d_params.u_cell_size,
+                                self.pseudo3d_params.v_cell_size,
+                            ],
+                            self.pseudo3d_params.depth_core,
+                            [self.pseudo3d_params.horizontal_padding] * 2
+                            + [self.pseudo3d_params.vertical_padding, 1],
+                            self.pseudo3d_params.expansion_factor,
+                        )[0]
+
+                        iter_workspace.remove_entity(receiver_entity.current_electrodes)
+                        iter_workspace.remove_entity(receiver_entity)
+
+                        xyz_out = mesh.centroids
+                        model_uids = deepcopy(models)
+                        for name, model in models.items():
+                            if model is None:
+                                continue
+                            elif isinstance(model, Data):
+                                model_values = weighted_average(
+                                    xyz_in, xyz_out, [model.values], n=1
+                                )[0]
+                            else:
+                                model_values = model * np.ones(len(xyz_out))
+
+                            model_object = mesh.add_data(
+                                {name: {"values": model_values}}
+                            )
+                            model_uids[name] = model_object.uid
+
+                        for key in ifile.data:
+                            param = getattr(self.pseudo3d_params, key, None)
+                            if hasattr(param, "uid"):
+                                if not isinstance(param, Data):
+                                    param.copy(
+                                        parent=iter_workspace, copy_children=True
```

Comment:
Why do we need to copy over all the originals? Feels like topo is enough here.

Reply:
We also need the data object, but I can do the two call separately from the loop. Done